From 22149342dfadeb129a6c15c8ed0750984a9ed41e Mon Sep 17 00:00:00 2001
From: Romain Perier <romain.perier@free-electrons.com>
Date: Thu, 8 Dec 2016 15:58:47 +0100
Subject: [PATCH 1761/2241] arm64: dts: marvell: Enable spi0 on the board
 Armada-3720-db

This commit enables the device node spi0 on the official development
board for the Marvell Armada 3700. It also adds sub-node for the 128Mb
SPI-NOR present on the board.

Change-Id: I8e709f009494a74fa41dfa91d6ed5b61e0535dbb
Signed-off-by: Romain Perier <romain.perier@free-electrons.com>
Tested-by: Gregory CLEMENT <gregory.clement@free-electrons.com>
Signed-off-by: Gregory CLEMENT <gregory.clement@free-electrons.com>
Signed-off-by: zachary <zhangzg@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/40887
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Victor Gu <xigu@marvell.com>
---
 arch/arm64/boot/dts/marvell/armada-3720-db.dts | 30 ++++++++++++++++++++++++++
 1 file changed, 30 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-3720-db.dts b/arch/arm64/boot/dts/marvell/armada-3720-db.dts
index 0e94950..12b146f 100644
--- a/arch/arm64/boot/dts/marvell/armada-3720-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3720-db.dts
@@ -63,6 +63,36 @@
 
 	soc {
 		internal-regs {
+			spi0: spi@10600 {
+				status = "okay";
+
+				m25p80@0 {
+					compatible = "jedec,spi-nor";
+					reg = <0>;
+					spi-max-frequency = <108000000>;
+					spi-rx-bus-width = <4>;
+					spi-tx-bus-width = <4>;
+
+					partitions {
+						compatible = "fixed-partitions";
+						#address-cells = <1>;
+						#size-cells = <1>;
+						partition@0 {
+							label = "bootloader";
+							reg = <0x0 0x200000>;
+						};
+						partition@200000 {
+							label = "U-boot Env";
+							reg = <0x200000 0x10000>;
+						};
+						partition@210000 {
+							label = "Linux";
+							reg = <0x210000 0xDF0000>;
+						};
+					};
+				};
+			};
+
 			i2c0: i2c@11000 {
 				status = "okay";
 				expander0: pca9555@22 {
-- 
2.7.4

